### ISSUE TITLE
`table-pipes` throws error on autoformated mdx v2 content

### ISSUE THREAD

**Issue by iAdramelk** [remarkjs/remark-lint#258]
### Your environment

- `remark-cli` – 8.0.1.
- `remark-mdx` – 2.0.0-next.8.
- `remark-lint-table-pipes` – 3.0.0.

### Steps to reproduce

`.remarkrc`

```
module.exports = {
  plugins: ["mdx", "lint-table-pipes"],
};
```

`markdown` content:

```
<Component>
| Head 1 | Head 2 |
| ------ | ------ |
| Cell 1 | Cell 2 |
</Component>
```

First run CLI with `-o` flag:

```
yarn remark 'path-to-file.mdx' -o
```

It will transform markdown to:

```
<Component>
  | Head 1 | Head 2 |
  | ------ | ------ |
  | Cell 1 | Cell 2 |
</Component>
```

Now run CLI again without `-0`:

```
yarn remark 'path-to-file.mdx'
```

### Expected behavior

Shouldn't throw an error.

### Actual behavior

```
  2:1  warning  Missing initial pipe in table fence  table-pipes  remark-lint
  4:1  warning  Missing initial pipe in table fence  table-pipes  remark-lint
```

Reason it breaks: https://github.com/remarkjs/remark-lint/blob/main/packages/remark-lint-table-pipes/index.js#L70

Looks like `offset` doesn't exclude padding inside the component.

**Comment by iAdramelk** [comment]
Tested with the latest cli and `remark-mdx` from the `master` branch of `mdx` repo (one with merged micromark update). This issue is is fixed.

**Comment by wooorm** [comment]
Nice! Will close this one then!
